libs/db/src/prisma/schema.prisma

```diff
@@ -374,6 +379,174 @@ model MapReviewComment {
+///     ranks for all slower runs on a leaderboard is an incredibly slow operation. (TODO: This hasn't been implemented yet).
+///     Instead, we index this way to allow us to very efficiently compute ranks per query using window functions. 
+model LeaderboardRun {
+  leaderboard Leaderboard @relation(fields: [mapID, gamemode, trackType, trackNum, style], references: [mapID, gamemode, trackType, trackNum, style], onDelete: Cascade)
+
+  user   User @relation(fields: [userID], references: [id], onDelete: Restrict)
+  userID Int
+
+  mmap  MMap @relation(fields: [mapID], references: [id], onDelete: Cascade)
+  mapID Int
+
+  gamemode  Int @db.SmallInt
+  trackType Int @db.SmallInt
+  trackNum  Int @db.SmallInt
+  style     Int @db.SmallInt
+
+  /// We don't store ticks in anticipation of future sub-tick system. If ticks 
+  /// are required, they can be calculated with `time * tickrate of gamemode`.
+  time Float @db.Real
+
+  /// We store this for LeaderboardRun and *not* PastRun for space efficiency.
+  stats Json /// run-stats.model.ts
+
+  // validated  Boolean // TODO: We'll be using this in future and blocking a bunch of stuff from DTO if false
+  replayHash String? @db.Char(40) /// SHA1
+
+  /// Flags are all the styles are run qualifies for. A styled run may qualify
+  /// for multiple leaderboards, and an separate entry is made for each. 
+  /// For example, a sideways surf run (for some track) would be added to the
+  /// sideways leaderboard and unstyled leaderboard. This field is used to "tag"
+  /// runs one leaderboard as also qualifying for another, e.g. the sideways 
+  /// run would get a "Sideways" tag on the unstyled leaderboard.
+  // This is an array because (a) it allows easier queries for specific flag(s)
+  // in Prisma (writing the paginated PastRuns get query raw would be really
+  // rough, though we could if absolutely needed), (b) we could conceivable
+  // exceed 64 flags at some point. This is of course a very large table where
+  // we care about size a great deal, though we expect the *majority* of runs
+  // If we need to cut back on the size of this table in the future, we could
+  // consider moving this to a bitfield (or adopting pguint for small values).
+  // But most runs submitted will not to have any flags, so it's probably okay.
+  flags Int[] @db.SmallInt
+
+  rank   Int  @db.Integer // TODO: SoonTM we'll stop storing this and just use a window function over the `time` column.
+  rankXP Int? @default(0) // TODO: And this shit will live in redis
+
+  createdAt DateTime @default(now())
+
+  pastRun   PastRun? @relation(fields: [pastRunID], references: [id])
+  pastRunID BigInt?  @unique
+
+  // Tempting to add a (userID, createdAt) index here for e.g. user profile
+  // page, but the PK index starting with `userID` will do some of the work
+  // anyway, proooobably not worth the space. Worth testing out on live DB in
+  // future!
```

Comment:
For this index specifically? I'm not sure how well that fits into a card since experimenting with indexes is something we should start doing fairly regularly once we've deployed and have a decently large dataset - we'll have analytics showing what queries are being used most often and what's taking the longest, so we should clone that DB from time to time and test performance with and without certain indexes. Exactly what index combinations we should test is a little hard to anticipate ahead of time so feel like a card would be a bit vague. We'll certainly not get the perfect combination of indexes the first time, we'll be tweaking this file frequently based on what seems to be performing well.